sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala

```diff
@@ -712,15 +831,84 @@ class WriteDistributionAndOrderingSuite
     }
   }
 
+  private def checkMicroBatchWriteRequirements(
+      tableDistribution: Distribution,
+      tableOrdering: Array[SortOrder],
+      tableNumPartitions: Option[Int],
+      expectedWritePartitioning: physical.Partitioning,
+      expectedWriteOrdering: Seq[catalyst.expressions.SortOrder],
+      writeTransform: DataFrame => DataFrame = df => df,
+      outputMode: String = "append",
+      expectAnalysisException: Boolean = false): Unit = {
+
+    catalog.createTable(ident, schema, Array.empty, emptyProps, tableDistribution,
+      tableOrdering, tableNumPartitions)
+
+    withTempDir { checkpointDir =>
+      val inputData = MemoryStream[(Long, String)]
+      val inputDF = inputData.toDF().toDF("id", "data")
+
+      val queryDF = outputMode match {
+        case "append" =>
+          inputDF
+        case "complete" =>
```

Comment:
Let's address "update" mode as well; it's ill-defined but there are users tolerating the definition due to the latency requirement.

Reply:
Sounds good. Will add.

Reply:
Now I remember why I did not add `update` initially. It is because `InMemoryTable` does not support it. Looks like it will be quite some work to make that happen as `InMemoryTable` has no notion of a key or something.

Any thoughts on the easiest way to test the update mode, @HeartSaVioR?

Reply:
As long as we have a semantic on "streaming update as append", it is same as append. We can revisit once we change the semantic.